classes/module/Module.php

```diff
@@ -425,6 +425,7 @@ public function install()
         $this->id = Db::getInstance()->Insert_ID();
 
         Cache::clean('Module::isInstalled' . $this->name);
+        Cache::clean('Module::getModuleIdByName_' . pSQL($this->name));
```

Comment:
@atomiix Can you explain how the cache becomes wrong?

Here is how the cache is being used
```
public static function getModuleIdByName($name)
    {
        $cache_id = 'Module::getModuleIdByName_' . pSQL($name);
        if (!Cache::isStored($cache_id)) {
            $result = (int) Db::getInstance()->getValue('SELECT `id_module` FROM `' . _DB_PREFIX_ . 'module` WHERE `name` = "' . pSQL($name) . '"');
            Cache::store($cache_id, $result);

            return $result;
        }

        return Cache::retrieve($cache_id);
    }
```

Why do we need to clear the cache when we install a module? I understand we might need to clear it when we upgrade or uninstall a module, but I would say "before a module is installed there is no cache entry"?

Reply:
@matks as soon as you call `getModuleIdByName()` you'll have a cache entry, also, when calling `isInstalled()` it will also call `getModuleIdByName()`...

Reply:
I see. I approve the PR.

Suggestion but it's out of scope: improve `getModuleIdByName()` so that it only registers cache entries for valid data 😆